components/AllocatedWorkers/DeallocateWorkerDetails/DeallocatedWorkerDetails.jsx

```diff
@@ -0,0 +1,97 @@
+import ErrorMessage from 'components/ErrorMessage/ErrorMessage';
+import { Button, TextArea } from 'components/Form';
+import Spinner from 'components/Spinner/Spinner';
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import Link from 'next/link';
+import { deleteAllocatedWorkers } from 'utils/api/allocatedWorkers';
+
+const DeallocatedWorkersDetails = ({
+  personId,
+  allocatedWorker,
+  firstName,
+  lastName,
+  workerType,
+  allocatedWorkerTeam,
+  isLastWorker,
+}) => {
+  const { register, handleSubmit } = useForm();
+  const [loading, setLoading] = useState(false);
+  const [complete, setComplete] = useState(false);
+  const patchData = {
+    worker: allocatedWorker,
+  };
+  const onSubmit = async (data) => {
+    setLoading(true);
+    try {
+      const test = JSON.stringify(Object.assign(patchData, data));
+      await deleteAllocatedWorkers(personId, test);
+    } catch {
+      <ErrorMessage label="Something went wrong, please retry"></ErrorMessage>;
+      console.log('Something went wrong');
+    }
+    setLoading(false);
+    setComplete(true);
```

Comment:
The `setComplete` should be inside the `try` otherwise you are showing the "complete message" also if the call fails